app/services/strategy_marketplace_service.py

```
@@ -665,45 +665,218 @@ async def get_user_strategy_portfolio(self, user_id: str) -> Dict[str, Any]:
+                    try:
+                        # Attempt async DB lookup for TradingStrategy by primary key
+                        from sqlalchemy import select
+                        
+                        strategy = None
+                        async for db in get_database():
+                            try:
+                                stmt = select(TradingStrategy).where(TradingStrategy.id == strategy_id)
+                                result = await db.execute(stmt)
+                                strategy = result.scalar_one_or_none()
+                                break
+                            except Exception as db_error:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)